TrackletAlgorithm/AllStubInnerMemory.h

```diff
@@ -333,6 +333,148 @@ class AllStubInner<6> : public AllStubInnerBase<DISK> // Can't generate cosim fi
 
 };
 
+// Data object definition
+template<>
+class AllStubInner<4> : public AllStubInnerBase<DISKPS>
+{
+  static_assert(DISKPS == 4, "DISKPS is assumed to be 4 in this class specialization.");
+public:
+  enum BitLocations {
+    // The location of the least significant bit (LSB) and most significant bit (MSB) in the AllStubInnerMemory word for different fields
+    kASFinePhiLSB = 0,
+    kASFinePhiMSB = kASFinePhiLSB + AllStubInnerBase<DISKPS>::kASFinePhiSize - 1,
+    kASIndexLSB = kASFinePhiMSB +1,
+    kASIndexMSB = kASIndexLSB + AllStubInnerBase<DISKPS>::kASIndexSize - 1,
+    kASBendLSB = kASIndexMSB + 1,
+    kASBendMSB = kASBendLSB + AllStubInnerBase<DISKPS>::kASBendSize - 1,
+    kASAlphaLSB = kASBendMSB + 1,
+    kASAlphaMSB = kASAlphaLSB + AllStubInnerBase<DISKPS>::kASAlphaSize - 1,
+    kASPhiLSB = kASAlphaMSB + 1,
+    kASPhiMSB = kASPhiLSB + AllStubInnerBase<DISKPS>::kASPhiSize - 1,
+    kASZLSB = kASPhiMSB + 1,
+    kASZMSB = kASZLSB + AllStubInnerBase<DISKPS>::kASZSize - 1,
+    kASRLSB = kASZMSB + 1,
+    kASRMSB = kASRLSB + AllStubInnerBase<DISKPS>::kASRSize - 1,
+    kASNegDiskLSB = kASRMSB + 1,
+    kASNegDiskMSB = kASNegDiskLSB + AllStubInnerBase<DISKPS>::kASNegDiskSize - 1
+  };
+  typedef ap_int<AllStubInnerBase<DISKPS>::kASRSize + 1> ASR;
+  typedef ap_int<AllStubInnerBase<DISKPS>::kASZSize> ASZ;
+  typedef ap_uint<AllStubInnerBase<DISKPS>::kASPhiSize> ASPHI;
+  typedef ap_uint<AllStubInnerBase<DISKPS>::kASAlphaSize> ASALPHA;
+  typedef ap_uint<AllStubInnerBase<DISKPS>::kASBendSize> ASBEND;
+  typedef ap_uint<AllStubInnerBase<DISKPS>::kASIndexSize> ASINDEX;
+  typedef ap_uint<AllStubInnerBase<DISKPS>::kASFinePhiSize> ASFINEPHI;
+  typedef ap_uint<AllStubInnerBase<DISKPS>::kAllStubInnerSize> AllStubInnerData;
+  typedef ap_uint<AllStubInnerBase<DISKPS>::kAllStubInnerSize-AllStubInnerBase<DISKPS>::kASIndexSize-AllStubInnerBase<DISKPS
+  >::kASFinePhiSize> AllStubData;
+  typedef ap_uint<1> ASNEGDISK; //FIXME this is always ap_uint<1> because csynth fails if this is ever 0 wide
+
+  // Constructors
+  AllStubInner(const AllStubInnerData& newdata):
+    data_(newdata)
+  {}
+
+ AllStubInner(const ASNEGDISK negdisk, const ASR r, const ASZ z, const ASPHI phi, const ASBEND bend, const ASINDEX index, const ASFINEPHI finephi):
```

Comment:
@mlarson02 Please scoot this line over to match the other constructors:
```suggestion
  AllStubInner(const ASNEGDISK negdisk, const ASR r, const ASZ z, const ASPHI phi, const ASBEND bend, const ASINDEX index, const ASFINEPHI finephi):
```